### PR TITLE
Configurable contact summary information

### PR DESCRIPTION
**Pull request by garethbowen**
#2914: allow for configuring additional cards to show up on contact pane.
#3037: Allow for configuring which fields show up in contact summary.
#2913: Allow for configuring additional context for showing/hiding create report actions.
#1930: Format phone numbers nicely

### CODE REVIEW

File: tests/karma/unit/services/db.js [hunk 1]
```diff
@@ -17,6 +17,7 @@ describe('DB service', function() {
     module(function ($provide) {
       $provide.factory('$window', function() {
         return {
+          angular: { callbacks: [] },
```
Comment (estellecomment) [inline tests/karma/unit/services/db.js]: whassat?

File: tests/nodeunit/unit/libphonenumber.js [hunk 2]
```diff
@@ -16,114 +16,114 @@ var settings = {
   default_country_code: COUNTRY_CODES.new_zealand
 };
 
-exports['format does nothing when already formatted'] = function(test) {
+exports['normalize does nothing when already normalizeted'] = function(test) {
```
Comment (estellecomment) [inline tests/nodeunit/unit/libphonenumber.js]: normalizeting is always a good thing

File: tests/nodeunit/unit/libphonenumber.js [hunk 3]
```diff
@@ -220,9 +220,9 @@ exports['validate returns false when funky punctuation in number'] = function(te
   test.done();
 };
 
-exports['validate returns true when extra zeros in international format'] = function(test) {
-  // Note : that's for NZ format. Would be nice to test specific formatting issues for target countries.
-  var actual = phonenumber.validate(settings, '+640275552636'); // Unnecessary leading 0 from domestic format
+exports['validate returns true when extra zeros in international normalize'] = function(test) {
```
Comment (estellecomment) [inline tests/nodeunit/unit/libphonenumber.js]: you did a find+replace in this file didn't you :D

File: static/js/services/contact-summary.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+      summary.cards = summary.cards || [];
+
+      summary.values.forEach(applyFilter);
+      summary.cards.forEach(function(card) {
+        card.values.forEach(applyFilter);
+      });
+      return summary;
+    };
+
+    return function(contact, reports) { // jshint ignore:line
```
Comment (garethbowen) [inline static/js/services/contact-summary.js]: Unused variables.

File: tests/karma/unit/services/db.js (hunk 1 above)
Comment (garethbowen) [inline tests/karma/unit/services/db.js]: IDK. It's an unrelated issue that I've cherry-picked to master too. I think angular must've updated something internally so mocking $window is even harder  :(

File: tests/nodeunit/unit/libphonenumber.js (hunk 2 above)
Comment (garethbowen) [inline tests/nodeunit/unit/libphonenumber.js]: Haha. Damn find and replace...

File: tests/nodeunit/unit/libphonenumber.js (hunk 3 above)
Comment (garethbowen) [inline tests/nodeunit/unit/libphonenumber.js]: How did you guess??

File: kanso.json [hunk 5]
```diff
@@ -317,6 +317,16 @@
                     }
                 }
             },
+            "contact_summary": {
+                "title": "Contact summary",
+                "type": "string",
+                "description": "A function to generate contact summaries, cards, and context objects.",
```
Comment (garethbowen) [inline kanso.json]: Good call. Fixed.

File: packages/libphonenumber/libphonenumber/utils.js [hunk 6]
```diff
@@ -2,44 +2,61 @@
...
+  try {
+    return _init(settings, phone).format(phonenumber.PhoneNumberFormat.E164);
+  } catch (e) {}
+  return false;
+};
+
+/**
+ * Returns the number formatted for display, or false if invalid.
+ */
 exports.format = function(settings, phone) {
```
Comment (garethbowen) [inline packages/libphonenumber/libphonenumber/utils.js]: `displayFormat` is a noun so doesn't work well for me. To me `format` and `normalize` definitely mean different things. Any other ideas?

File: static/js/filters/phone.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+    var settings;
+
+    Settings().then(function(result) {
+      settings = result;
+    });
+
+    var format = function(phone) {
+      if (settings) {
+        return libphonenumber.format(settings, phone);
```
Comment (garethbowen) [inline static/js/filters/phone.js]: Good catch. Fixed.

File: static/js/services/contact-summary.js [hunk 8]
```diff
@@ -0,0 +1,53 @@
+var SETTING_NAME = 'contact_summary';
```
Comment (garethbowen) [inline static/js/services/contact-summary.js]: Added.

File: static/js/services/contact-summary.js [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+    var getScript = function() {
+      return Settings().then(function(settings) {
+        return settings[SETTING_NAME];
+      });
+    };
+
+    var applyFilter = function(value) {
+      if (value.filter) {
+        try {
+          value.value = $filter(value.filter)(value.value);
```
Comment (garethbowen) [inline static/js/services/contact-summary.js]: Done.

**Pull request merged**